seqfold/seqfold.go

```diff
@@ -0,0 +1,157 @@
+package seqfold
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/TimothyStiles/poly/checks"
+)
+
+type Comp map[byte]byte
+
+type MultiBranchEnergies struct {
+	A, B, C, D float64
+}
+
+// Energy holds two energies, enthaply and entropy
+// SantaLucia & Hicks (2004), Annu. Rev. Biophys. Biomol. Struct 33: 415-440
+type Energy struct {
+	// enthaply
+	H float64
+	// entropy
+	S float64
+}
+
+type BpEnergy map[string]Energy
+
+type LoopEnergy map[int]Energy
+
+type Energies struct {
+	BULGE_LOOPS     LoopEnergy
+	COMPLEMENT      Comp
+	DE              BpEnergy
+	HAIRPIN_LOOPS   LoopEnergy
+	MULTIBRANCH     MultiBranchEnergies
+	INTERNAL_LOOPS  LoopEnergy
+	INTERNAL_MM     BpEnergy
+	NN              BpEnergy
+	TERMINAL_MM     BpEnergy
+	TRI_TETRA_LOOPS BpEnergy
+}
+
+type Interval struct {
```

Comment:
what does this interval type do? What are I and J? Start and end of subsequence?

Reply:
Yes, AFAIU it is basically a subsequence that holds an inner/child structure, I'm open to suggestion for the naming.
Something like:

```go
type Subsequence struct{
    Start, End int
}
```